rdm/0007-access-control.md

```diff
@@ -0,0 +1,222 @@
+- Start Date: 2019-10-01
+- RFC PR: [#7](https://github.com/inveniosoftware/rfcs/pull/7)
+- Authors: Pablo Panero
+
+# Access Control
+
+## Summary
+
+Overall design for management of access control and permissions over resources in Invenio RDM.
+
+## Motivation
+
+- As an administrator, I want to have both public and restricted records, so that not all the records are publicly visible.
+- As a user, I want to restrict access to files in a public record, so that I can safely store and cite my dateset.
+- As a user, I want to restrict files and/or metadata during an embargo period, so that I can comply with publisher rules.
+- As an administrator, I want to have a flexible and easy way of defining access control policies, so that I do not need to code them.
+- As an administrator, I want to have IP-based access-control, so that I can comply with publisher rules.
+- As a user, I want to be able to obtain a sharable link, so that my colleagues can access the files of my restricted record.
+- As an administrator, I want to easily define roles like *academic stuff* based on my IdP (Identity Provider), so that I can implement bussiness rules.
+- As a user, I want to share a preview of a deposit record prior to publishing it, so that my colleagues can validate it.
+- As a user, I want to grant access to my restricted records by listing specific users and/or groups, so that I can easily revoke the access if needed.
+- As an administrator, I want to predefine roles, so that I can easily keep QA and production accurately reflect each other??
+- As an administrator, I want to restrict access to certain records based on user features (e.g. registered user, member of a specific community, member of academia), so that I can implement my institutions policies.
+- As a user, I want to access my data even after I have left my university.
+- As an administrator, I want to open my repositories for other institutions and their IDMS (?? Institutional Data Management System).
+
+## Terminology
+
+- **Resources**: The term resource in the system represents anything you want to protect. Resources can both be at the object-level (e.g. access to a record) or of more general nature (e.g access to the administration interface).
+Actions: Resources can be protected by requiring the ability to perform one or more actions. 
```

Comment:
hmm... I think the terminology is still an action. Since we can have `Action('read')`. However, in implementation this is translated to *needs* and *excludes*.

@fenekku what do you think?